modules/gapi/samples/pipeline_modeling_tool/dummy_source.hpp

```diff
@@ -14,45 +14,84 @@ class DummySource final: public cv::gapi::wip::IStreamSource {
 public:
     using Ptr = std::shared_ptr<DummySource>;
     DummySource(const double       latency,
-                const OutputDescr& output);
+                const OutputDescr& output,
+                const bool         drop_frames);
     bool pull(cv::gapi::wip::Data& data) override;
     cv::GMetaArg descr_of() const override;
 
 private:
     double  m_latency;
     cv::Mat m_mat;
-    using TimePoint =
-        std::chrono::time_point<std::chrono::high_resolution_clock>;
-    cv::optional<TimePoint> m_prev_pull_tp;
+    bool    m_drop_frames;
+    double  m_next_tick_ts = -1;
+    int64_t m_curr_seq_id  = 0;
 };
 
 DummySource::DummySource(const double       latency,
-                         const OutputDescr& output)
-    : m_latency(latency) {
+                         const OutputDescr& output,
+                         const bool         drop_frames)
+    : m_latency(latency), m_drop_frames(drop_frames) {
     utils::createNDMat(m_mat, output.dims, output.precision);
     utils::generateRandom(m_mat);
 }
 
 bool DummySource::pull(cv::gapi::wip::Data& data) {
     using namespace std::chrono;
     using namespace cv::gapi::streaming;
+
     // NB: In case it's the first pull.
-    if (!m_prev_pull_tp) {
-        m_prev_pull_tp = cv::util::make_optional(high_resolution_clock::now());
+    if (m_next_tick_ts == -1) {
+        m_next_tick_ts = utils::timestamp<milliseconds>() + m_latency;
```

Comment:
Please put a comment that we're intentionally waiting for m_latency at first frame here

Reply:
Added